src/Features/CSharp/Portable/ConvertProgram/ConvertProgramTransform_ProgramMain.cs

```diff
@@ -56,8 +63,15 @@ public static async Task<Document> ConvertToProgramMainAsync(Document document,
             return document;
         }
 
-        private static async Task<SyntaxNode> GenerateProgramClassAsync(
+        private static bool IsProgramClass(ClassDeclarationSyntax declaration)
+        {
+            return declaration.Identifier.Text == WellKnownMemberNames.TopLevelStatementsEntryPointTypeName &&
+                   declaration.Modifiers.Any(SyntaxKind.PartialKeyword);
```

Comment:
```suggestion
                   declaration.Modifiers.Any(SyntaxKind.PartialKeyword) &&
                   !declaration.Modifiers.Any(SyntaxKind.FileKeyword);
```

Reply:
Just checked, `file` modifier doesn't break program from compiling/running:
```cs
file class Program
{
    private static void Main(string[] args)
    {
        Console.WriteLine("Hello, World!");
    }
}
```
This will write `Hello, World!` to the console as expected

Reply:
@DoctorKrolic My bad. I thought that the following will generate two separate classes:

```csharp
System.Console.WriteLine();

file partial class Program { }
```

It might be worth adding a test anyway.